packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -466,6 +449,38 @@ export function useConnections({
     ]
   );
 
+  const oidcAttemptConnectNotifyDeviceAuth = useCallback(
+    (
+      connectionInfo: ConnectionInfo,
+      {
+        verificationUrl,
+        userCode,
+      }: { verificationUrl: string; userCode: string },
+      signal: AbortSignal
+    ) => {
+      // For single connection store the device auth info so that we can append
+      // it to the "Connecting..." modal
+      dispatch({
+        type: 'oidc-attempt-connect-notify-device-auth',
+        connectionInfo,
+        verificationUrl,
+        userCode,
+      });
+
+      // For multiple connection we show a special type of the confirmation modal
```

Comment:
Should this be behind a condition checking for multiple connections? Or is that handled somewhere else because it looks like we might end up with two modals in single connection.

Reply:
It's handled consistently for all toasts and modals in a single place where they are defined, I can maybe remove the comment if it's confusing, wdyt?

Reply:
yea I think that'd be helpful or maybe a small note about the conditional handling logic happening elsewhere.